tortoisebeacon/tortoise_beacon.go

```diff
@@ -613,25 +608,27 @@ func (tb *TortoiseBeacon) runConsensusPhase(ctx context.Context, epoch types.Epo
 	ticker := time.NewTicker(tb.votingRoundDuration + tb.weakCoinRoundDuration)
 	defer ticker.Stop()
 
-	tb.sendFollowingVotesLoopIteration(ctx, epoch, firstRound+1)
+	for round := firstRound; round <= tb.lastPossibleRound(); round++ {
+		go tb.startWeakCoin(ctx, epoch, round)
+		go func(epoch types.EpochID, round types.RoundID) {
+			if err := tb.sendVotes(ctx, epoch, round); err != nil {
+				tb.Log.With().Error("Failed to send voting messages",
+					log.Uint64("epoch_id", uint64(epoch)),
```

Comment:
It's not a scope of your PR, but as `types.EpochID` is `uint32` now, and as you're touching this, it can be changed to `uint32`
```suggestion
					log.Uint32("epoch_id", uint32(epoch)),
```